discovery/aws/ec2.go

```diff
@@ -243,21 +224,23 @@ func (d *EC2Discovery) refresh(ctx context.Context) ([]*targetgroup.Group, error
 		})
 	}
 
+	// Only refresh the AZ ID map if we have never been able to build one.
+	// Prometheus requires a reload if AWS adds a new AZ to the region.
+	if d.azToAZID == nil {
+		if err := d.refreshAZIDs(ctx); err != nil {
+			level.Warn(d.logger).Log(
```

Comment:
I would have this as debug, and add the information about ec2:DescribeAvailabilityZones in the documentation.